packages/application/src/test/application.spec.ts

```diff
@@ -0,0 +1,266 @@
+        },
+        language: {
+          title: 'Language',
+          description: 'Code snippet implementation language',
+          type: 'string',
+          uihints: {
+            field_type: 'dropdown',
+            default_choices: ['Python', 'Java', 'R', 'Scala', 'Markdown']
+          },
+          minLength: 1
+        },
+        code: {
+          title: 'Code',
+          description: 'Code snippet code lines',
+          type: 'array',
+          uihints: {
+            field_type: 'code'
+          }
+        }
+      },
+      required: ['language', 'code']
+    }
+  },
+  required: ['schema_name', 'display_name', 'metadata']
+};
+
+const codeSnippetMetadata = {
+  schema_name: 'code-snippet',
+  display_name: 'tester',
+  name: 'tester',
+  metadata: {
+    language: 'Python',
+    code: ['hello_world']
+  }
+};
+
+const updatedCodeSnippetMetadata = {
+  schema_name: 'code-snippet',
+  display_name: 'tester',
+  name: 'tester',
+  metadata: {
+    language: 'Python',
+    code: ['testing']
+  }
+};
+
+beforeAll(async () => {
+  await server.start();
+});
+
+afterAll(async () => {
+  await server.shutdown();
+});
+
+describe('@elyra/application', () => {
+  describe('FrontendServices', () => {
+    describe('#getSchema', () => {
+      it('should get schema', async () => {
+        expect(await FrontendServices.getSchema('code-snippets')).toEqual([
+          codeSnippetSchema
+        ]);
+      });
+    });
+    describe('#getAllSchema', () => {
+      it('should get all schema', async () => {
+        const schemas = await FrontendServices.getAllSchema();
+        expect(schemas).toHaveLength(3);
+      });
+    });
+    describe('metadata requests', () => {
+      it('should create metadata instance', async () => {
+        expect(
+          await FrontendServices.postMetadata(
+            'code-snippets',
+            JSON.stringify(codeSnippetMetadata)
+          )
+        ).toMatchObject(codeSnippetMetadata);
+      });
+
+      it('should get the correct metadata instance', async () => {
+        expect(
+          await FrontendServices.getMetadata('code-snippets')
+        ).toMatchObject([codeSnippetMetadata]);
+      });
+
+      it('should update the metadata instance', async () => {
+        expect(
+          await FrontendServices.putMetadata(
+            'code-snippets',
+            'tester',
+            JSON.stringify(updatedCodeSnippetMetadata)
+          )
+        ).toMatchObject(updatedCodeSnippetMetadata);
+      });
+
+      it('should delete the metadata instance', async () => {
+        await FrontendServices.deleteMetadata('code-snippets', 'tester');
+        const snippets = await FrontendServices.getMetadata('code-snippets');
+        expect(snippets).toHaveLength(0);
+      });
+    });
+  });
+
+  describe('RequestHandler', () => {
+    describe('#makeGetRequest', () => {
+      it('should make get request', async () => {
+        expect(
+          await RequestHandler.makeGetRequest(
```

Comment:
We have a service layer that performs some of these requests, shouldn't we test that instead of testing the backend services?

Reply:
Are you talking about `FrontendServices`? I'm testing both because there are a few instances outside of `application` that call `RequestHandler` directly (for example [here](https://github.com/elyra-ai/elyra/blob/69ba69722502c2b9de1cda3a17be369022f0f008/packages/pipeline-editor/src/PipelineService.tsx#L98))